proparse/src/main/java/org/prorefactor/proparse/antlr4/PreproEval.java

```diff
@@ -1,33 +1,423 @@
+   * 
+   * @return A Boolean object
+   */
+  @Override
+  public Boolean visitPreproIfEval(PreproIfEvalContext ctx) {
+    LOGGER.trace("Entering visitPreproIfEval()");
+    Object o = visit(ctx.expr());
+    LOGGER.trace("Exiting visitPreproIfEval() with return value '{}'", o);
+    return (o != null) && getBool(o);
+  }
+
+  // ****
+  // Expr
+  // ****
+  @Override
+  public Object visitAndOr(AndOrContext ctx) {
+    boolean b1 = getBool(visit(ctx.expr(0)));
+    boolean b2 = getBool(visit(ctx.expr(1)));
+
+    if (ctx.op.getType() == PreprocessorParser.OR) {
+      return b1 || b2;
+    } else {
+      return b1 && b2;
+    }
+  }
+
+  @Override
+  public Object visitStringOp(StringOpContext ctx) {
+    Object o1 = visit(ctx.expr(0));
+    Object o2 = visit(ctx.expr(1));
+
+    if (ctx.op.getType() == PreprocessorParser.MATCHES) {
+      return matches(o1, o2);
+    } else {
+      return getString(o1).toLowerCase().startsWith(getString(o2).toLowerCase());
+    }
+  }
+
+  @Override
+  public Object visitComparison(ComparisonContext ctx) {
+    Object o1 = visit(ctx.expr(0));
+    Object o2 = visit(ctx.expr(1));
+
+    switch (ctx.op.getType()) {
+      case PreprocessorParser.EQ:
+      case PreprocessorParser.EQUAL:
+        return compare(o1, o2, Compare.EQ);
+      case PreprocessorParser.GTORLT:
+      case PreprocessorParser.NE:
+        return compare(o1, o2, Compare.NE);
+      case PreprocessorParser.RIGHTANGLE:
+      case PreprocessorParser.GTHAN:
+        return compare(o1, o2, Compare.GT);
+      case PreprocessorParser.LEFTANGLE:
+      case PreprocessorParser.LTHAN:
+        return compare(o1, o2, Compare.LT);
+      case PreprocessorParser.GTOREQUAL:
+      case PreprocessorParser.GE:
+        return compare(o1, o2, Compare.GE);
+      case PreprocessorParser.LTOREQUAL:
+      case PreprocessorParser.LE:
+        return compare(o1, o2, Compare.LE);
+      default:
+        return null;
+    }
+  }
+
+  @Override
+  public Object visitPlus(PlusContext ctx) {
+    if (ctx.op.getType() == PreprocessorParser.PLUS)
+      return opPlus(visit(ctx.expr(0)), visit(ctx.expr(1)));
+    else
+      return opMinus(visit(ctx.expr(0)), visit(ctx.expr(1)));
+  }
+
+  @Override
+  public Object visitMultiply(MultiplyContext ctx) {
+    switch (ctx.op.getType()) {
+      case PreprocessorParser.STAR:
+      case PreprocessorParser.MULTIPLY:
+        return opMultiply(visit(ctx.expr(0)), visit(ctx.expr(1)));
+      case PreprocessorParser.SLASH:
+      case PreprocessorParser.DIVIDE:
+        return opDivide(visit(ctx.expr(0)), visit(ctx.expr(1)));
+      case PreprocessorParser.MODULO:
+        Double m1 = getFloat(visit(ctx.expr(0))) + .5;
+        Double m2 = getFloat(visit(ctx.expr(1))) + .5;
+        return new Integer(m1.intValue() % m2.intValue());
+      default:
+        return null;
+    }
+  }
+
+  @Override
+  public Object visitNot(NotContext ctx) {
+    return new Boolean(!getBool(visit(ctx.expr())));
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove this "Boolean" constructor [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS2129)